website/docs/attached-class.md

```diff
@@ -0,0 +1,75 @@
+---
+id: attached-class
+title: T.attached_class
+---
+
+`T.attached_class` can be used to refer to the type of instances from a
+singleton-context, in a way that is friendly to inheritance.
+
+```ruby
+class AbstractData
```

Comment:
This class is called `Abstract` but it's not actually `abstract!`. You might want to just change the name so that it doesn't say `Abstract`, because I'd rather not have people think that being Abstract is a requirement to use attached_class.